Pod/Classes/SeafConnection.m

```diff
@@ -1247,66 +1224,73 @@ - (void)checkPhotos:(BOOL)force
      [self performSelectorInBackground:@selector(backGroundCheckPhotos:) withObject:[NSNumber numberWithBool:force]];
 }
 
-- (void)backGroundCheckPhotos:(NSNumber *)forceNumber
-{
+- (void)backGroundCheckPhotos:(NSNumber *)forceNumber {
     bool force = [forceNumber boolValue];
     SeafDir *uploadDir = _syncDir;
     bool shouldSkip = !_inAutoSync || (!force && [self photosInSyncing] > 0) || (self.firstTimeSync && !uploadDir);
     if (shouldSkip) {
         return;
     }
-
+    
     @synchronized(self) {
         if (_inCheckPhotoss) return;
         _inCheckPhotoss = true;
     }
-
-    Debug("Check photos for server %@, current %u %u", _address, (unsigned)_photosArray.count, (unsigned)_uploadingArray.count);
-
+    
```

Comment:
This functions gets a little overstaffed, could you break it into a few smaller functions.
e.g. buildAutoSyncPredicte, filterOutUploadedPhotos